### PR TITLE
Add E tag to NIP-10

### LINKED ISSUE #812: NIP-10 markers problem
I want to nag about `e` tag markers. The markers convention is really bad for clients that want to lazily load replies instead of loading everything at once. Motivation is saving bandwidth.

One way of fixing it ~without using markers~:
**edit**: after @vitorpamplona comments, keeping everything as-is but also just adding an uppercase `E` tag referencing the event you are replying to (for both root replies or reply replies) would be enough. For example, clients could lazy load direct root replies with this: `["REQ", "dkldsaj", { kinds: [1], #E: ["<root id>"], limit: 5 }]`

~- `e` and `a` reference the root event (just to keep compatibility with existing clients that load every thread message at once; we can keep the "root" marker so that existing clients don't change code)~
~- `E` and `A` reference the reply-to event and will have the same value of the lowercase versions when user is replying to the root message (lazy loading clients can use these tags)~
~- mentions use NIP-21 nostr:nevent1... OR, if you really want a tag, use a new tag such as `M`. **Edit:** Or better yet, use existing `q` (quote)  tag~

@jb55 @vitorpamplona @staab and other client authors please consider this when creating new messages. What do you think?

### PR DESCRIPTION
**Pull request by arthurfranca**
Fix #812

### CODE REVIEW

File: 10.md [hunk 1]
```diff
@@ -50,8 +50,10 @@ Those marked with `"reply"` denote the id of the reply event being responded to.
 
 A direct reply to the root of a thread should have a single marked "e" tag of type "root".
 
->This scheme is preferred because it allows events to mention others without confusing them with `<reply-id>` or `<root-id>`.
+All reply events must also add a capital letter `E` tag, without marker, set to the id of the event they are replying too. This tag
```
[the remark this answers is not in the document]
Comment (arthurfranca) [reply 10.md]: thanks, fixed it

**Comment by jb55** [comment]
I think it would make sense for your app to be backwards compatible instead of forcing everyone to update to your streaming model. Then we could make this a recommendation for clients that know they they will be creating large threads.

**Comment by arthurfranca** [comment]
A bit disappointing to know you are against it mostly because your app won't use it but I respect your honesty.

I will just clarify that my intention isn't to force clients to change how they load threads. Adding a tag wouldn't accomplish that. I simply want to make lazy loading threads possible and for that I asked for cooperation. Adding the `E` tag without removing anything would make room for both approaches.

**Comment by jb55** [comment]
On Wed, Oct 18, 2023 at 08:02:45PM -0700, arthurfranca wrote:
>A bit disappointing to know you are against it mostly because your app
>won't use it but I respect your honesty.
>
>I will just clarify that my intention isn't to force clients to change
>how they load threads. Adding a tag wouldn't accomplish that. I simply
>want to make lazy loading threads possible and for that I asked for
>cooperation. Adding the `E` tag without removing anything would make
>room for both approaches.

Keep in mind the ecosystem is very large at this point, if you have to
convince 100+ clients and microapps to update their code to support your
use case, then your approach is wrong. I am not against it mainly
because "my app won't use it", I'm trying to help you to understand your
approach won't work at scale, especially once nostr is much larger. If
damus had to do this everytime someone wanted some core protocol thing
changed to support their usecase, then the NIPs would be horribly
complicated.

I almost want to fork the NIPs repo and remove all of the extra stuff
people have been adding to NIPs I've written (custom reactions, zap
splits). I would hate to see nostr go the way of the browser where
people keep adding to existing NIPs until it's too confusing to build
new clients in a weekend.

Every time you extend a NIP its hardforking a subset of the protocol,
increasing implementation spec divergence and confusion.

**Comment by vitorpamplona** [comment]
Forking the repo is a terrible take. Yes, we want to keep Nostr simple but removing stuff from the NIP repo doesn't make implementations disappear. You should follow your own advice to @arthurfranca. If you have to convince other apps to stop implementing things they already do to support your vision for Nostr, your approach is wrong.

**Comment by arthurfranca** [comment]
I had to close the PR because Damus has a large user base. Without it, clients relying on the tag would miss all these users' replies. Smaller clients that don't adhere wouldn't be a big deal and possibly most would update their clients to the edited NIP-10 guidelines sooner or later.

I think @fiatjaf downvoted too because usually lower case `e` tag is used for direct replies. For example, I've seen events without root marker, just with single `e` tag with `reply` marker. Fundamentally, the `root` marker is the offending guy that ideally would've been simply another tag like `o` (original poster).

The `mention` marker is already being replaced by the `q` (quote) tag which is great, although @jb55 introduced it ignoring NIP-10 spec, how come uh? heheh Guess rules are a bit relaxed after all.

**Comment by fr4nzap** [comment]
Focusing on the querying side may be the right approach after all. Given the bazaar implementation approach, having flexibility in queries is a must for nostr.

 I'm not suggesting it's easy but would allow many of these features without having to ask all implementations to start writing data differently (and have the benefit of backward-compatibility)

**Comment by mikedilger** [comment]
> I would hate to see nostr go the way of the browser where people keep adding to existing NIPs until it's too confusing to 
build new clients in a weekend.

It is far beyond that already.  I'm 1 year in and it turns out I'm not even compliant with NIP-01 regarding 'a' tags.

**Comment by weex** [comment]
It would be great to see more unmerged NIPs be merged and enter `final` state. That way it becomes clearer when a NIP is no longer up for change and that instead energy should be put into a successor. Seems to me, the first step is merging a lot of unmerged `drafts`, then pinging authors to see which can be made `final`.

**Pull request closed without merging**